code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardUiProvider.kt

```diff
@@ -0,0 +1,148 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging.contentcards
+
+import com.adobe.marketing.mobile.AdobeCallbackWithError
+import com.adobe.marketing.mobile.AdobeError
+import com.adobe.marketing.mobile.Messaging
+import com.adobe.marketing.mobile.aepcomposeui.contentprovider.AepUIContentProvider
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepUITemplate
+import com.adobe.marketing.mobile.messaging.MessagingConstants
+import com.adobe.marketing.mobile.messaging.Proposition
+import com.adobe.marketing.mobile.messaging.SchemaType
+import com.adobe.marketing.mobile.messaging.Surface
+import com.adobe.marketing.mobile.services.Log
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+class ContentCardUiProvider(val surfaceString: String) : AepUIContentProvider {
+    companion object {
+        private const val SELF_TAG: String = "MessageAssetDownloader"
+    }
+
+    private val _contentFlow = MutableStateFlow<List<AepUITemplate>>(emptyList())
+    private val contentFlow: StateFlow<List<AepUITemplate>> = _contentFlow
+
+    override suspend fun getContent(): Flow<List<AepUITemplate>> {
+        CoroutineScope(Dispatchers.IO).launch {
+            val surface = Surface(surfaceString)
+            val surfaceList = mutableListOf<Surface>()
+            surfaceList.add(surface)
+            Messaging.updatePropositionsForSurfaces(surfaceList)
+            Messaging.getPropositionsForSurfaces(surfaceList) {
+                getCardsForSurface(surfaceList) {
+                    it.onSuccess { templateList ->
+                        _contentFlow.value = templateList
+                    }
+                    it.onFailure { error ->
+                        Log.error(MessagingConstants.LOG_TAG, SELF_TAG, "Failed to get content: ${error.message}")
+                        _contentFlow.value = emptyList()
+                    }
+                }
+            }
+        }
+        return contentFlow
+    }
+
+    fun getCardsForSurface(
+        surfaceList: MutableList<Surface>,
```

Comment:
Why do we need a separate `surfaceList` here? The class construction already accepts a surface

Reply:
Messaging update and get proposition needs List<Surface> as input, so need to wrap one surface as list as well.

Reply:
Surface is a location in the app where cards are to be displayed. While it makes sense to pass a list to prefetch and cache multiple surfaces in updatePropositionsForSurface since it makes the actual network call, why would customers need to get more than a single surface while actually displaying? Wouldn't they display one location at a time? How will they handle a list?

Reply:
get Cards for surface is not public API, renamed to getAepUIForSurfaceList, we are using this function internally.
ContentCardUiProvider takes a single surface as parameter. Then getContent() call is placed by user, we are
we are wrapping this surface as a single element in list. as Messaeging's get and update proposition requires List

Reply:
changed to getAepUITemplateList which is not public API, it is internal function.
getAepUi() and getContent() are two available public APIs

Reply:
Hmm I still don't get why we need to accept `surfaceList` here. `surfaceString` is passed in when the class is created. Why can't we `surfaceString` directly instead of accepting another list here?

Reply:
Done